lib/symboldatabase.cpp

```diff
@@ -1524,11 +1463,14 @@ void SymbolDatabase::createSymbolDatabaseIncompleteVars()
             tok = tok->link();
             continue;
         }
-        if (Token::Match(tok, "catch|typeid (")) {
+        if (tok->isCpp() && (Token::Match(tok, "catch|typeid (") ||
+                             Token::Match(tok, "static_cast|dynamic_cast|const_cast|reinterpret_cast"))) {
```

Comment:
This does not check for the opening (`<`) unlike the previous condition - is that intentional? Could this be replaced with `isCast()`?

Reply:
There was no explicit handling of C++ casts (and the stuff between <...>) in `createSymbolDatabaseIncompleteVars()`, so I wonder how that worked.
isCast() is set for `(`, so that's too late for the target type.
We should also reject a C++ cast keyword not followed by `<` as a syntax error.